src/core/bsequence.ml

```diff
@@ -0,0 +1,49 @@
+
+type +'a node =
+    | Nil
+    | Cons of 'a * 'a t
+and 'a t = unit -> 'a node
+
+let empty () = Nil
+
+let return x () = Cons(x, empty)
```

Comment:
I've renamed this `make` in `bopt` and `bresult`. On the fence about that, but I don't think `return` is a very good name.